Makefile.dev

```diff
@@ -32,31 +32,13 @@ dev/camlp4.dbg:
 revision:
 	$(SHOW)'CHECK revision'
 	$(HIDE)rm -f revision.new
-ifeq ($(CHECKEDOUT),svn)
-	$(HIDE)set -e; \
-        if test -x "`which svn`"; then \
-	  export LC_ALL=C;\
-	  svn info . | sed -ne '/URL/s/.*\/\([^\/]\{1,\}\)/\1/p' > revision.new; \
-	  svn info . | sed -ne '/Revision/s/Revision: \([0-9]\{1,\}\)/\1/p'>> revision.new; \
-	fi
-endif
-ifeq ($(CHECKEDOUT),gnuarch)
-	$(HIDE)set -e; \
-        if test -x "`which tla`"; then \
-	  LANG=C; export LANG; \
-	  tla tree-version  >  revision.new ; \
-	  tla tree-revision | sed -ne 's|.*--||p' >> revision.new ; \
-	fi
-endif
 ifeq ($(CHECKEDOUT),git)
 	$(HIDE)set -e; \
 	if test -x "`which git`"; then \
 	  LANG=C; export LANG; \
 	  GIT_BRANCH=$$(git branch -a | sed -ne '/^\* /s/^\* \(.*\)/\1/p'); \
-	  GIT_HOST=$$(hostname); \
-	  GIT_PATH=$$(pwd); \
-	  (echo "$${GIT_HOST}:$${GIT_PATH},$${GIT_BRANCH}") > revision.new; \
-	  (echo "$$(git log -1 --pretty='format:%H')") >> revision.new; \
+	  (echo "$$(git describe --tags | sed 's/^V//')") > revision.new; \
+	  (echo "$${GIT_BRANCH}") >> revision.new; \
```

Comment:
Currently, the `$CHECKEDOUT` variable is set by a function in `configure.ml`. Given that this PR is dropping support for gnuarch and svn (which is fine), I would consider removing this `vcs` function entirely and test if `.git` exists here in the Makefile.

It should also be possible to create a small independent shell script that is called from the `revision` target in the Makefile but can also be called stand-alone without running `./configure` first. This separation would be useful because the script creating the `revision` file should also be called from the script generating the source tarball if we want it to display the correct version.